lib/src/main_common.dart

```diff
@@ -3,17 +3,16 @@ import 'package:allthenews/src/app/app_flavors.dart';
 import 'package:allthenews/src/di/injector.dart';
 import 'package:allthenews/src/ui/common/theme/theme_notifier.dart';
 import 'package:allthenews/src/ui/pages/presentation/presentation_notifier.dart';
-import 'package:firebase_crashlytics/firebase_crashlytics.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
 
-void mainCommon(Environment flavor) {
+Future<void> mainCommon(Environment flavor) async {
+  WidgetsFlutterBinding.ensureInitialized();
   injectDependencies(flavor);
-  FlutterError.onError = Crashlytics.instance.recordFlutterError;
   runApp(
     MultiProvider(
       providers: [
-        ChangeNotifierProvider(create: (_) => inject<ThemeNotifier>()),
+        ChangeNotifierProvider.value(value: inject<ThemeNotifier>()),
```

Comment:
jak theme notifier będzie tworzony przy użyciu fabryki to z powrotem na create

Reply:
tak z ciekawości, wiesz może co się stanie jak dasz `value` kiedy będzie fabryka? :P

Reply:
to raczej jest tak, że jak masz już obiekt stworzony to powinno się go dać do value. Jak my to wstrzykujemy to przy fabryce wydaje mi się, że nie ma żadnego znaczenia czy będzie value czy create
![image](https://user-images.githubusercontent.com/24854180/101816569-78b4b380-3b21-11eb-8b4c-48b0fe268c7a.png)

Reply:
no właśnie ciekaw byłem jak on to traktuje, bo przy value i fabryce tworzysz źle z tego korzystasz bo tworzysz nową instancję i zacząłem się zastanawiać czym to skutkuje w drzewie, ale to jak będzie mi się nudzić to kiedyś przeanalizuję :D